supportedBackends/tes/src/test/scala/cromwell/backend/impl/tes/TesAsyncBackendJobExecutionActorSpec.scala

```diff
@@ -0,0 +1,115 @@
+package cromwell.backend.impl.tes
+
+import common.mock.MockSugar
+import cromwell.core.logging.JobLogger
+import cromwell.core.path
+import cromwell.filesystems.blob.{BlobFileSystemManager, BlobPath, WSMBlobSasTokenGenerator}
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import java.nio.file.Path
+import scala.util.{Failure, Success, Try}
+
+class TesAsyncBackendJobExecutionActorSpec extends AnyFlatSpec with Matchers with MockSugar {
+  behavior of "TesAsyncBackendJobExecutionActor"
+
+  val fullyQualifiedName = "this.name.is.more.than.qualified"
+  val workflowName = "mockWorkflow"
+  val someBlobUrl = "https://lz813a3d637adefec2c6e88f.blob.core.windows.net/sc-d8143fd8-aa07-446d-9ba0-af72203f1794/nyxp6c/tes-internal/configuration/supported-vm-sizes"
+  val someNotBlobUrl = "https://www.google.com/path/to/exile"
+  var index = 0
+
+  val blobInput_0 = Input(
+    name = Option(fullyQualifiedName + "." + index),
+    description = Option(workflowName + "." + fullyQualifiedName + "." + index),
+    url = Option(someBlobUrl),
+    path = someBlobUrl,
+    `type` = Option("FILE"),
+    content = None
+  )
+  index = index+1
+
+  val blobInput_1 = Input(
+    name = Option(fullyQualifiedName + "." + index),
+    description = Option(workflowName + "." + fullyQualifiedName + "." + index),
+    url = Option(someBlobUrl),
+    path = someBlobUrl,
+    `type` = Option("FILE"),
+    content = None
+  )
+  index = index+1
+
+  val notBlobInput_1 = Input(
+    name = Option(fullyQualifiedName + "." + index),
+    description = Option(workflowName + "." + fullyQualifiedName + "." + index),
+    url = Option(someNotBlobUrl + index),
+    path = someNotBlobUrl + index,
+    `type` = Option("FILE"),
+    content = None
+  )
+  index = index+1
+
+  val notBlobInput_2 = Input(
+    name = Option(fullyQualifiedName + "." + index),
+    description = Option(workflowName + "." + fullyQualifiedName + "." + index),
+    url = Option(someNotBlobUrl + index),
+    path = someNotBlobUrl + index,
+    `type` = Option("FILE"),
+    content = None
+  )
+
+  // Mock blob path functionality.
+  val mockWsmEndpoint = "https://wsm.mock.com/endpoint"
+  val mockWorkspaceId = "e58ed763-928c-4155-0000-fdbaaadc15f3"
+  val mockContainerResourceId = "e58ed763-928c-4155-1111-fdbaaadc15f3"
+
+  val mockLogger: JobLogger = mock[JobLogger]
+  val mockBlobPath: BlobPath = mock[BlobPath]
+  val mockTokenGenerator: WSMBlobSasTokenGenerator = mock[WSMBlobSasTokenGenerator]
+  val mockFsm: BlobFileSystemManager = mock[BlobFileSystemManager]
```

Comment:
It feels unusual to me to mock these and then share them among the tests. I would normally create all mocks anew for each test, and make a function that creates and returns them if it turns out to be a lot of boilerplate. 

I don't know for a fact that this doesn't work, it just feels dangerous given that there's mutable state in there and we don't know what order tests are run in.